spark/src/main/scala/org/apache/spark/sql/delta/storage/dv/DeletionVectorStore.scala

```diff
@@ -205,22 +205,36 @@ class HadoopFileSystemDVStore(hadoopConf: Configuration)
       // Lazily create the writer for the deletion vectors, so that we don't write an empty file
       // in case all deletion vectors are empty.
       private var outputStream: FSDataOutputStream = _
+      private var writtenBytes = 0L
 
       override def write(data: Array[Byte]): DeletionVectorStore.DVRangeDescriptor = {
         if (outputStream == null) {
           val overwrite = false // `create` Java API does not support named parameters
           outputStream = path.fs.create(path.path, overwrite)
           outputStream.writeByte(DeletionVectorStore.DV_FILE_FORMAT_VERSION_ID_V1)
+          writtenBytes += 1
         }
         val dvRange = DeletionVectorStore.DVRangeDescriptor(
           offset = outputStream.size(),
           length = data.length,
-          checksum = DeletionVectorStore.calculateChecksum(data)
-          )
+          checksum = DeletionVectorStore.calculateChecksum(data))
+
+        if (writtenBytes != dvRange.offset) {
+          recordDeltaEvent(
+            deltaLog = null,
```

Comment:
Do we have a lot of call sites for createWriter? I wonder if we can pass the delta log there.

Reply:
That would change more places in the code, which is what I don't want two.
Having the DV path and class name is more than enough for debugging.